### PR TITLE
Use tools.go instead of retool

### PR DESCRIPTION
**Pull request by loxp**
UCP #2182 
Signed-off-by: loxp <loxppqqxol@gmail.com>

### What is changed and how it works?
Use tools.go instead of retool

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
    - make
    - make basic-test

Code changes

Side effects

- .retool/ need to be removed manually.

Related changes

- Document

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,22 +9,24 @@ BASIC_TEST_PKGS := $(filter-out $(INTEGRATION_TEST_PKGS),$(TEST_PKGS))
 PACKAGES := go list ./... | grep -v 'dashboard/uiserver'
 PACKAGE_DIRECTORIES := $(PACKAGES) | sed 's|$(PD_PKG)/||'
 GOCHECKER := awk '{ print } END { if (NR > 0) { exit 1 } }'
-RETOOL := ./scripts/retool
 OVERALLS := overalls
 
-FAILPOINT_ENABLE  := $$(find $$PWD/ -type d | grep -vE "(\.git|\.retools)" | xargs ./scripts/retool do failpoint-ctl enable)
-FAILPOINT_DISABLE := $$(find $$PWD/ -type d | grep -vE "(\.git|\.retools)" | xargs ./scripts/retool do failpoint-ctl disable)
+GOBIN := $(shell pwd)/.gobin
```
Comment (HunDunDM) [inline Makefile]: I think it can be placed in the `./bin` directory. @nolouch , what do you think?

File: go.mod [hunk 2]
```diff
@@ -42,19 +42,18 @@ require (
 	github.com/pingcap/sysutil v0.0.0-20200206130906-2bfa6dc40bcd
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.0.0
-	github.com/sirupsen/logrus v1.2.0
-	github.com/spf13/cobra v0.0.3
-	github.com/spf13/pflag v1.0.1
+	github.com/sirupsen/logrus v1.4.2
```
Comment (HunDunDM) [inline go.mod]: It looks like the `logrus` upgrade caused CI to fail.

**Comment by HunDunDM** [comment]
Sorry, the lower version of `golangci-lint` seems to have a lot of problems. You can wait for #2195 and remove it.

File: Makefile [hunk 3]
```diff
@@ -9,22 +9,24 @@ BASIC_TEST_PKGS := $(filter-out $(INTEGRATION_TEST_PKGS),$(TEST_PKGS))
...
+PATH := $(GOBIN):$(PATH)
+
+FAILPOINT_ENABLE  := $$(find $$PWD/ -type d | grep -vE "(\.git|\.retools)" | xargs failpoint-ctl enable)
```
Comment (rleungx) [inline Makefile]: Can we remove all retools related things?

Comment (loxp) [reply Makefile]: It can be removed. But if an older version is used (retool is existed), is it need to skip these pkgs?

File: go.mod (hunk 2 above)
Comment (loxp) [reply go.mod]: Yes, I downgrade logrus now.

File: Makefile (hunk 3 above)
Comment (rleungx) [reply Makefile]: Actually, there is no fail point in `.retools` packages.

File: Makefile [hunk 4]
```diff
@@ -93,18 +95,17 @@ basic-test:
...
 	  --enable errcheck \
 	  $$($(PACKAGE_DIRECTORIES))
-	CGO_ENABLED=0 ./scripts/retool do gosec $$($(PACKAGE_DIRECTORIES))
+	CGO_ENABLED=0 gosec $$($(PACKAGE_DIRECTORIES))
 
 check-all: static lint tidy
 	@echo "checking"
 
-retool-setup: export GO111MODULE=off
+retool-setup: export GO111MODULE=on
```
Comment (rleungx) [inline Makefile]: How about `install-tools`?

Comment (loxp) [reply Makefile]: OK

File: Makefile (hunk 3 above)
Comment (loxp) [reply Makefile]: OK. And remote `.retools` in DEADLOCK_ENABLE and DEADLOCK_DISABLE?

Comment (rleungx) [reply Makefile]: I think we can.

Comment (loxp) [reply Makefile]: There's no `vendor`.

File: Makefile (hunk 1 above)
Comment (nolouch) [reply Makefile]: I prefer to `.tools/bin` or `.tools`, replace  the `.retools`.

**Pull request merged**